src/test/java/org/zalando/baigan/e2e/configs/SomeConfiguration.java

```diff
@@ -11,6 +12,7 @@ public interface SomeConfiguration {
     SomeConfigObject someConfig();
     String someValue();
     Boolean isThisTrue();
+    Boolean toggleFlag(CustomContextProvider customContextProvider,CustomContextProvider secondProvider);
```

Comment:
Have you considered just passing a `Context` object with a context name and value or directly a `Map`? This could be varargs to pass arbitrary contexts and not having to pass any if none is desired. What is the added value of having the `ContextProvider`?

It would basically put the `ContextProvider::getContextParam` call to the user, but would also make them more flexible in how they want to generate the context.

Reply:
Well, okay. I'm not fully sold on that point. For your example, other solutions would work equally well. However, I don't want to block this here as I agree with moving into a direction that simplifies the whole thing. If @lukasniemeier-zalando has no objections, I'm fine with this step.